### ISSUE TITLE
Incorrect resource file paths when a .qrc file is in a subdirectory

### ISSUE THREAD

**Issue by torokati44** [mesonbuild/meson#3112]
When building an application using the `qt5` module, if a `.qrc` file is in a subdirectory, then the paths for the files referenced in it are not reconstructed correctly.

See this simple application for reproduction: [testapp.zip](https://github.com/mesonbuild/meson/files/1741814/testapp.zip)

After the usual ritual of running `meson builddir`, then changing into `builddir` and running `ninja`, I get this error:

> ninja: error: '../qtgui/qtgui/img/squiggle.png', needed by 'qtgui/qt5-images_qrc.cpp', missing and no known rule to make it

Note the incorrectly duplicated `qtgui` in the path.

I think this might have been introduced by https://github.com/mesonbuild/meson/commit/58d68f3d10dcfa34ed9ce04e2e5aa81a70a8feca.

I was able to "fix" it by changing this line in the `parse_qrc` function in `qt.py`:
```python
                    result.append(os.path.join(state.subdir, relative_part, child.text))
```
Into this:
```python
                    result.append(os.path.join(relative_part, child.text))
```

I suspect that the `state.subdir` part is also added by `meson` automatically.

The reason I didn't submit a PR is that I'm not at all sure if this is a proper solution in the general case, or if this breaks for other applications. However, if anyone with more familiarity with the `qt5` module could confirm that this would indeed be a good fix, I'd be happy to do so. (Although it might not be that much more helpful for a single line change...)

**Comment by leiflm** [comment]
#2438 might be related?

**Comment by torokati44** [comment]
It definitely is. I found #2438, as well as #2315 before. However, seeing that one breaks the tests, and the other is closed unmerged, I didn't really know what to do with them.

Are you suggesting that finishing one of those PRs properly would solve my problem too (which I think is likely)? And that this is maybe something I could take on doing?

**Comment by nirbheek** [comment]
This should already be fixed with https://github.com/mesonbuild/meson/pull/2956. Can you try with the 0.44.1 stable release and master?

**Comment by torokati44** [comment]
It is indeed solved in 0.44.1. I was unaware of this .1 release, and thought that with 0.44.0 I'm using the latest version, so I did not look at the closed PRs. I am sorry to bother you unnecessarily.

If you feel the need to do so, you are free to use my little `testapp` in addition to the test in the referenced PR, although they look almost identical to me.

**Comment by nirbheek** [comment]
It was just released yesterday, so I don't blame you for not knowing about it ;)
